src/origins.rs

```diff
@@ -779,13 +785,24 @@ impl AddressOriginSet {
                 }
             }
             for origin in item {
-                let tal_metrics = &mut tal_metrics_vec[origin.tal_index];
-                tal_metrics.roas += 1;
+                let tal_metrics = &mut metrics.tals[origin.tal_index].vrps;
+                let mut repo_metrics = match origin.repository_index {
+                    Some(index) => Some(&mut metrics.repositories[index].vrps),
+                    None => None,
+                };
                 for addr in origin.addrs {
-                    tal_metrics.total_valid_vrps += 1;
+                    tal_metrics.valid += 1;
+                    if let Some(vrps) = repo_metrics.as_mut() {
+                        vrps.valid += 1
+                    }
+                    metrics.vrps.valid += 1;
```

Comment:
Aren't these counted twice (here, and finally when the metrics are collapsed?)
Goes for the other metrics in this part as well.

Reply:
The engine collapses only the publication metrics while here only VRP-related metrics are collected. I have stuck a note on the collapse function to point this out.